processor/file.go

```diff
@@ -206,18 +178,13 @@ func walkDirectoryParallel(root string, output chan *FileJob) {
 
 			shouldSkip := false
 
-			if gitIgnoreError == nil && gitIgnore.Match(fpath, false) {
-				if Verbose {
-					printWarn("skipping file due to git ignore: " + f.Name())
-				}
-				shouldSkip = true
-			}
-
-			if ignoreError == nil && ignore.Match(fpath, false) {
-				if Verbose {
-					printWarn("skipping file due to ignore: " + f.Name())
+			for _, i := range ignores {
```

Comment:
Given how these loops seem to come as a pair with `for _, exclude := range excludes`, could we unify the two?

It should be possible to define a `FileMatcher` interface, and get both (wrapped) `regexp` matchers and `gitignore` matchers in the same list.

Reply:
That might be one to do when I get around to adding the multiple .gitignore support.

I want to keep them separate for the moment at least till that's done because its implementation is already doing my head in :)

Reply:
BTW if you want to help rebuild godirwalk let me know. Will want to make it as fast as possible, run in parallel without having to built in the consumer libraries and allow some sort of pre-flight so we can request it to let us know if .gitignore files exist before processing so it can be loaded for futher processing.

Reply:
Sure, I'm happy to help with that. I've thought on and off about handling non-linear streams of work which don't map well to `chan`s (e.g. walking directories and finding more sub-directories in the process). I've got some ideas I want to play around with, I'll probably put something simple/generic up on my github when I get something to work.

Reply:
Neat. If its fast (IE almost as fast) as whats there it would be a good thing to switch over to. Those recursive .gitignore files are a bitch with godirwalk.